src/services/userService.js

```diff
@@ -165,51 +134,50 @@ async function editUserRole(idOfUserForEdit, requestBody, requesterId) {
     const userForEdit = await User.findById(idOfUserForEdit);
     const requester = await User.findById(requesterId);
     const isAdmin = requester.role == 'admin' ? true : false;
-    if (!isAdmin) {
+
+    if (!isAdmin || requester.isDeleted || !requester.isApproved) {
         throw new Error('You do not have rights to modify the record!');
     }
-    if (!userForEdit || !requester) {
+
+    if (!userForEdit) {
         throw new Error('User not found');
     }
-    if (userForEdit.isDeleted || requester.isDeleted) {
-        throw new Error('This profile is deleted!');
-    }
-    if (!userForEdit.isApproved || !requester.isApproved) {
-        throw new Error('This profile is not approved!');
-    }
 
     if (requestBody.role == 'organizer') {
-        if (userForEdit.organizatorName == '' && userForEdit.phone == '') {
-            if (!requestBody.organizatorName || !requestBody.phone) {
+        if (userForEdit.organizatorName == '') {
+            if (!requestBody.organizatorName) {
                 throw new Error('Fill all required fields!');
```

Comment:
More specified errors -> "Organizator's name is missing" for example

Reply:
added